src-electron/generator/matter/app/zap-templates/common/ClustersHelper.js

```diff
@@ -591,8 +591,9 @@ function enhancedCommands(commands, types) {
       // The 'response' property contains the response returned by the `selectCommandById`
       // helper. But this one does not contains all the metadata informations added by
       // `enhancedItem`, so instead of using the one from ZAP, retrieve the enhanced version.
+      const clusterName = command.clusterName;
       command.response = commands.find(
-        (command) => command.name == responseName
+        (command) => command.name == responseName && command.clusterName == clusterName
```

Comment:
Before merging changes like these it will be good to see the Matter regen on this change before merging it in but I don't think there is any way to test a Matter branch with this zap branch before a merge anymore.

Reply:
There is no delta on a vanilla Matter tree.

On a tree with https://github.com/project-chip/connectedhomeip/pull/24507 applied, without this change the wrong LauncherResponse is used and the generated code does not compile.

The relevant diff (from correct to incorrect, since that's easier for me to generate) is:
```
diff --git a/zzz_generated/chip-tool/zap-generated/test/Commands.h b/zzz_generated/chip-tool/zap-generated/test/Commands.h
index 4dd8631090..1638b128c2 100644
--- a/zzz_generated/chip-tool/zap-generated/test/Commands.h
+++ b/zzz_generated/chip-tool/zap-generated/test/Commands.h
@@ -53766,7 +53766,8 @@ private:
                 chip::app::Clusters::ApplicationLauncher::Commands::LauncherResponse::DecodableType value;
                 VerifyOrReturn(CheckDecodeValue(chip::app::DataModel::Decode(*data, value)));
                 VerifyOrReturn(CheckValue("status", value.status, 0U));
-                VerifyOrReturn(CheckValueAsString("data", value.data, chip::ByteSpan(chip::Uint8::from_const_char("data"), 4)));
+                VerifyOrReturn(CheckValuePresent("data", value.data));
+                VerifyOrReturn(CheckValueAsString("data.Value()", value.data.Value(), chip::CharSpan("data", 4)));
             }
             break;
         case 4:
@@ -53775,7 +53776,8 @@ private:
                 chip::app::Clusters::ApplicationLauncher::Commands::LauncherResponse::DecodableType value;
                 VerifyOrReturn(CheckDecodeValue(chip::app::DataModel::Decode(*data, value)));
                 VerifyOrReturn(CheckValue("status", value.status, 0U));
-                VerifyOrReturn(CheckValueAsString("data", value.data, chip::ByteSpan(chip::Uint8::from_const_char("data"), 4)));
+                VerifyOrReturn(CheckValuePresent("data", value.data));
+                VerifyOrReturn(CheckValueAsString("data.Value()", value.data.Value(), chip::CharSpan("data", 4)));
             }
             break;
         case 5:
@@ -53784,7 +53786,8 @@ private:
                 chip::app::Clusters::ApplicationLauncher::Commands::LauncherResponse::DecodableType value;
                 VerifyOrReturn(CheckDecodeValue(chip::app::DataModel::Decode(*data, value)));
                 VerifyOrReturn(CheckValue("status", value.status, 0U));
-                VerifyOrReturn(CheckValueAsString("data", value.data, chip::ByteSpan(chip::Uint8::from_const_char("data"), 4)));
+                VerifyOrReturn(CheckValuePresent("data", value.data));
+                VerifyOrReturn(CheckValueAsString("data.Value()", value.data.Value(), chip::CharSpan("data", 4)));
             }
             break;
         default:
```

Note that it's using the wrong types, because one of the LauncherResponse commands has an optional CHAR_STRING field named Data and the other has a required OCTET_STRING one, and the wrong response command is being picked up.

Reply:
This helps but will be good to see the Matter CI pass after this change. It will be useful to have that for bigger changes.